File: Kotlin/1.two_sum.kt
fun main() {
    println(twoSum(intArrayOf(2,7,11,15), 9).joinToString())
    println(twoSum(intArrayOf(3,2,4), 6).joinToString());
    println(twoSum(intArrayOf(3,3), 6).joinToString());
}

fun twoSum(nums: IntArray, target: Int): IntArray {
    val map = mutableMapOf<Int, Int>();

    for(i in nums.indices) {
        map[nums[i]] = i;
    }

    for(i in nums.indices) {
        val i2 = map[target - nums[i]];

        if(i2 !=null && i != i2) {
            return intArrayOf(i, i2);
        }
    }

    throw Exception("Not found!")
}

File: Dart/1.two_sum.dart
void main() {
  print(twoSum([2, 7, 11, 15], 9));
  print(twoSum([3, 2, 4], 6));
  print(twoSum([3, 3], 6));
}

List<int> twoSum(List<int> nums, int target) {
  final map = {};

  for (int i = 0; i < nums.length; i++) {
    map[nums[i]] = i;
  }

  for (int i = 0; i < nums.length; i++) {
    final i2 = map[target - nums[i]];

    if (i2 != null && i != i2) {
      return [i, i2];
    }
  }

  throw Exception("Not found!");
}
